src/avl.c

```diff
@@ -514,39 +518,35 @@ bool search_BFS_avl(avl_pp root, int val, bool stop)
 	if (!enqueue(queue, *root)) {
 		log(ERROR, "enqueue failed!\n");
 		destroy_queue(queue);
-		return FALSE;
+		return -1;
 	}
 
 	/* Loop through all nodes in the Queue */
 	while ((node = dequeue(queue)) != NULL) {
-		log(INFO, "tracking...\n");
+		//log(INFO, "tracking...\n");
 
 		/* Process left child of node */
 		if (node->left) {
-			if (node->left->data == val) {
-				log(INFO, "FOUND %d\n", val);
-				ret = TRUE;
-
-				if (stop)
-					break;
+			if (node->left->keyNode == key) {
+				//log(INFO, "FOUND %d\n", val);
+				destroy_queue(queue);
+				return node->left->data;
```

Comment:
Please don't remove the `stop` functionality. It was useful for printing the whole tree. If your tree is sorted by key, print the key and the value.